common/SC_Win32Utils.cpp

```diff
@@ -52,12 +52,18 @@ void win32_ExtractContainingFolder(char* folder, const char* pattern, int maxCha
         folder[0] = 0;
 }
 
-void win32_GetKnownFolderPath(int folderId, char* dest, int size) {
-    // Use a temporary buffer, as SHGetFolderLocation() requires it
-    // to be at least MAX_PATH size, but destination size may be less
-    char buf[MAX_PATH];
-    SHGetFolderPath(NULL, folderId, NULL, 0, buf);
-    strncpy(dest, buf, size);
+std::wstring win32_StringToWstring(const std::string& string) {
+    DWORD size = MultiByteToWideChar(CP_UTF8, 0, string.c_str(), -1, nullptr, 0);
```

Comment:
What happens if the string is empty? 

Cpp ref says this
> If empty() returns true, the pointer is a non-null pointer that should not be dereferenced.

Reply:
Arguably, both these functions are not doing any checking, neither whether the input string is valid, nor whether the conversion succeeded. 

Here, do you suggest I'd do something like 

```cpp
DWORD size = MultiByteToWideChar(CP_UTF8, 0, string.empty() ? nullptr : string.c_str(), -1, nullptr, 0);
```
?